lib/bloc_record/collection.rb

```diff
@@ -0,0 +1,51 @@
+module BlocRecord
+  class Collection < Array
+    def update_all(updates)
+      ids = self.map(&:id)
+      self.any? ? self.first.class.update(ids, updates) : false
+    end
+
+    def group(*args)
+      ids = self.map(&:id)
+      self.any? ? self.first.class.group_by_ids(ids, args) : false
+    end
+
+    def distinct
+      new_collection = self
```

Comment:
This will not create a new collection. However, you do have everything laid out to make this work. You just need to not focus on pulling duplicates out of the current collection, but rather make sure duplicates do not enter the new collection.